internal/vpn.go

```diff
@@ -121,10 +123,22 @@ func initVpn(app *App, client *http.Client, crypto CryptoHandler) error {
 		if err != nil {
 			return fmt.Errorf("Unable to generate private key: %s", err)
 		}
-		log.Printf("Uploading Wireguard pub key(%s).", pub)
-		if err := updateConfig(app, client, pub); err != nil {
-			os.Remove(wgPriv)
-			return fmt.Errorf("Unable to server config with VPN public key: %s", err)
+		log.Printf("Uploading Wireguard public key(%s).", pub)
+		for x := 1; x <= 10; x++ {
+			if err := updateConfig(app, client, pub); err != nil {
+				log.Println("Unable to upload Wireguard public key:", err)
+				if x < 10 {
+					log.Printf("Sleeping %d seconds and trying again\n", x)
+					time.Sleep(time.Duration(x) * time.Second)
+				} else {
+					os.Remove(wgPriv)
+					// There's no point running if we can't communicate with
+					// the server. Let systemd restart us and hope for the best.
+					os.Exit(1)
```

Comment:
Should we instead have this logic as part of main and then some retries for when uploading data?

Just thinking as we start adding similar commands/features to fioconfig.

Reply:
this approach is just forcing the initialization logic to not proceed if we can't talk to the server and then eventually die which would cause systemd to restart us.

another approach we might do instead is to add some sort of "work queue" to the app. then the main loop could execute items in the work queue at each loop. this could be a different way we try to upload this data. I *think* this work-queue idea is more what you have in mind. I'll give that a shot.

Reply:
Yeah, that would work better, and we could even not abort the execution as this will for sure cause noise on the system level, specially because this is an expected scenario (no network for a long period).

And it would even allow us to extend later with additional handlers, so all positive.

Reply:
@ricardosalveti - i just force pushed a different approach. I think this wound up being a bit cleaner.